ADVANCED.md

```diff
@@ -488,7 +480,9 @@ The advantage of extracting the prop types is that you won't need to export ever
 import { ComponentProps, JSXElementConstructor } from 'react';
 
 // goes one step further and resolves with propTypes and defaultProps properties
-type ApparentComponentProps<C> = C extends JSXElementConstructor<infer P>
+type ApparentComponentProps<
+  C extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>
```

Comment:
Required for `ComponentProps`. There's probably an argument to be had to remove the constraint in `@types/react` and return `never` if the constraint isn't fulfilled.